### PR TITLE
Documentation on building doc locally

### LINKED ISSUE #8274: Add instructions for building documentation locally for documentation related contributions
### What should we add?

I can't find instructions for building documentation locally for qiskit-terra in https://qiskit.org/documentation/contributing_to_qiskit.html#contributing-to-documentation nor https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md I think it would to add something similar to https://github.com/Qiskit/qiskit-machine-learning/issues/413 for qiskit-terra so that community contributors can follow to make documentation related contributions

### PR DESCRIPTION
**Pull request by uchoi91**
### Summary
In Contributing doc, I have added section for building documentation locally using tox, sphinx-build, and make html.

### Details and comments
fix #8274

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -259,6 +260,30 @@ Note that in order to run `python setup.py ...` commands you need have build
...
+to build documentation locally to review.
+The easiest and recommended way to build the documentation is to use tox. Once you run the command, the output will be located at `docs/_build/html`
+
+using [**tox**](https://tox.readthedocs.io/en/latest/#):
+
+```
+tox -edocs
+```
+
+using [**sphinx-build](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
```
Comment (ElePT) [inline CONTRIBUTING.md]: sphinx-build is missing a `**`:
Screenshot 2023-10-23 at 10 56 15
```suggestion
using [**sphinx-build**](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -259,6 +260,30 @@ Note that in order to run `python setup.py ...` commands you need have build
...
+
+```
+sphinx-build -b html docs docs/_build/html
+```
+
+using make html:
```
Comment (ElePT) [inline CONTRIBUTING.md]:
```suggestion
using ``make html``:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ contributing to terra, these are documented below.
 * [Changelog generation](#changelog-generation)
 * [Release Notes](#release-notes)
 * [Installing Qiskit Terra from source](#installing-qiskit-terra-from-source)
+* [Building Documentation Locally](#building-documentation-locally)
```
Comment (ElePT) [inline CONTRIBUTING.md]: Most of the titles don't capitalize all words (the ones that do, refer to concepts such as "Release Notes" or names such as "Lint"), so I'd change the capitalization to:

```suggestion
* [Building documentation locally](#building-documentation-locally)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -259,6 +260,30 @@ Note that in order to run `python setup.py ...` commands you need have build
 dependency packages installed in your environment, which are listed in the
 `pyproject.toml` file under the `[build-system]` section.
 
+## Building Documentation Locally
```
Comment (ElePT) [inline CONTRIBUTING.md]:
```suggestion
## Building documentation locally
```

**Review by ElePT — COMMENTED** [review]
Thanks for this addition to the guidelines! I think this can be very helpful for our contributors. I have only left a few minor suggestions and then it will look good to me.

**Comment by uchoi91** [comment]
@ElePT Thank you so much for suggestions! I've pushed out the changes.

**Review by AngeloDanducci — APPROVED** [review]

**Comment by uchoi91** [comment]
@ElePT could you review this PR when you get a chance? Thanks!

**Review by ElePT — DISMISSED** [review]
@uchoi91 I have fixed the merge conflicts and updated the title to be a bit more specific (API docs instead of documentation). The change looks good to me but it would be better to also have @Eric-Arellano's approval to make sure it makes sense from the docs perspective.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -580,6 +580,25 @@ rather than via `tox`. If you have installed the development packages in your py
...
+The easiest and recommended way to build the documentation is to use tox. 
+Once you run any of these command, the output will be located at `docs/_build/html`
+
+1. using [**tox**](https://tox.readthedocs.io/en/latest/#):
+
+```
+tox -edocs
+```
+
+2. using [**sphinx-build**](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
```
Comment (Eric-Arellano) [inline CONTRIBUTING.md]: @ElePT why document this? People can always figure it out if they want. My concern with documenting it is that it makes it a "blessed" path, which somewhat implies that we're going to commit to this working. 

It's valuable to have only one single endorsed way to do builds. If others want to do their custom thing, go ahead, but it's on them to verify it works properly.

For example, right now the below command is not actually the same as what we run with Tox:

https://github.com/Qiskit/qiskit/blob/7e73fcf51e1d3557e0f291bb757248ba14d39ebe/tox.ini#L81

And it doesn't install the optional requirements:

https://github.com/Qiskit/qiskit/blob/7e73fcf51e1d3557e0f291bb757248ba14d39ebe/tox.ini#L79

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -580,6 +580,25 @@ rather than via `tox`. If you have installed the development packages in your py
 `pip install -r requirements-dev.txt`, then `ruff` and `black` will be available and can be run from
 the command line. See [`tox.ini`](tox.ini) for how `tox` invokes them.
 
+## Building API docs locally
+
+If you have made changes to the API documentation, you can run the commands below
+to build documentation locally to review the html output. 
+The easiest and recommended way to build the documentation is to use tox. 
+Once you run any of these command, the output will be located at `docs/_build/html`
```
Comment (Eric-Arellano) [inline CONTRIBUTING.md]:
```suggestion
Once you run any of these command, the output will be located at `docs/_build/html`.
Then, open up the file `index.html` in your browser.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -580,6 +580,25 @@ rather than via `tox`. If you have installed the development packages in your py
...
+```
+tox -edocs
+```
+
+2. using [**sphinx-build**](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
+
+```
+sphinx-build -b html docs docs/_build/html
+```
+
```
Comment (Eric-Arellano) [inline CONTRIBUTING.md:598]:
```suggestion
Sometimes Sphinx can get in a bad cache state. Run `tox -e docs-clean`
to reset Sphinx's cache.
```

**Review by Eric-Arellano — COMMENTED** [review]
Thanks for the contribution!

File: CONTRIBUTING.md (hunk 5 above)
Comment (ElePT) [reply CONTRIBUTING.md]: That is a really good point and really well sold :) I was originally ok with the idea of showing another path but hadn't thought about committing to it working. Your suggestion below about clearing the cache also covers the case I had in mind (which was tox getting stuck).

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -580,6 +580,27 @@ rather than via `tox`. If you have installed the development packages in your py
...
+tox -edocs
+```
+
+2. using [**sphinx-build**](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
+
+```
+sphinx-build -b html docs docs/_build/html
+```
+Sometimes Sphinx can get in a bad cache state. Run `tox -e docs-clean`
+to reset Sphinx's cache.
```
Comment (Eric-Arellano) [inline CONTRIBUTING.md:597]: My bad - I didn't add a blank line before and after this text.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -580,6 +580,27 @@ rather than via `tox`. If you have installed the development packages in your py
...
+```
+tox -edocs
+```
+
+2. using [**sphinx-build**](https://www.sphinx-doc.org/en/master/man/sphinx-build.html):
+
+```
+sphinx-build -b html docs docs/_build/html
+```
+Sometimes Sphinx can get in a bad cache state. Run `tox -e docs-clean`
```
Comment (ElePT) [inline CONTRIBUTING.md:596]:
```suggestion

Sometimes Sphinx can get in a bad cache state. Run `tox -e docs-clean`
```

**Review by ElePT — APPROVED** [review]

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**